deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml

```diff
@@ -405,6 +405,12 @@ runcmd:
   # Run installation tests
   - python3 /opt/build/check_installation.py || die "Installation check failed!";
 
+  # Remove Linux omsagent before generalising the image
+  - echo ">=== $(date +%s) Removing Linux omsagent ===<"
+  - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh
+  - sh onboard_agent.sh --purge || die "Could not remove Linux omsagent!";
```

Comment:
Can we work out what this actually does? In general, I don't like downloading and running arbitrary code from the internet (even if we *do* trust Microsoft more than an average provider).

If it's just deleting some files/directories, could we do it ourselves?

Reply:
I did also wonder about this, but figured, on balance, it's from Microsoft directly and would be the best way to be sure everything that's supposed to be uninstalled is uninstalled.

The first thing it does is download the agent and runs `omsagent-1.17.2-0.universal.x64.sh` with the `--purge` flag, which runs https://github.com/microsoft/OMS-Agent-for-Linux/blob/master/tools/purge_omsagent.sh, which runs https://github.com/microsoft/OMS-Agent-for-Linux/blob/master/installer/scripts/uninstall.sh.

Now, most of that looks like, for our case, it's just a lot of code dedicated to detecting what OS you're running and eventually removing some files/directories, but it's still maybe a bit more than I'm happy to replicate. There are quite a lot of conditionals in there that I'd have to go through to make sure they're not important. Which I can do.

Reply:
Thoughts @JimMadge ?

Reply:
Just some more exploration - the `purge_omsagent.sh` is there already, without downloading the bundle. But running it simply tells you to uninstall the agent through the portal, since the `uninstall.sh` script is not there. There is an `uninstall` bin, which does some of the work. But the extension is still there on the portal. Trying to install the agent using `Setup_SRE_Monitoring` then just hangs, saying it is removing the misconfigured extension.

I think there is some secret sauce that we are missing without going through all the steps in the script, as the script seems to clean it out completely and allow proper installation of the agent. We might be able to download the `uninstall.sh` script alone and use that, but at that point we might as well download the bundle script as MS suggest.

Reply:
Also not keen to maintain this ourselves to work around Microsoft distributing this as a "pipe curl to sudo bash" type script.

The only thing I would say is to see if we can specify a particular version/tag/commit instead of always fetching from master.
The same approach we take with certain actions _etc._.

Reply:
Have now edited to use the most recent commit of that specific file.